dvc/commands/experiments/apply.py

```diff
@@ -4,13 +4,21 @@
 from dvc.cli import completion
 from dvc.cli.command import CmdBase
 from dvc.cli.utils import append_doc_link
+from dvc.ui import ui
 
 logger = logging.getLogger(__name__)
 
 
 class CmdExperimentsApply(CmdBase):
     def run(self):
-        self.repo.experiments.apply(self.args.experiment, force=self.args.force)
+        if not self.args.force:
+            ui.write(
+                "The --no-force option is deprecated and will be removed in a future"
+                " DVC release. To revert the result of 'exp apply', run:\n"
+                "\n\tgit reset --hard\n"
+                "\tgit stash apply refs/exps/apply/stash"
+            )
```

Comment:
The suggested commands here are placeholder until we decide on `exp revert`

Reply:
Since we are mostly leaving the behavior as is, can we drop this deprecation warning, or do you still want to deprecate `--no-force` for some other reason?

Should we show the warning whenever there are changes overwritten? So try `--no-force`, and if it fails, try `--force` and show the warning?

Reply:
We can no longer get git conflict errors with the new behavior. We now do a checkout from a workspace that is always clean (since we stash the changes) and we no longer try to `git merge` the stashed changes back onto the new checkout afterwards. While we do manually restore the workspace untracked files which did not exist in the experiment, those files can also never conflict with anything.

Basically from git's perspective we are never overwriting anything now, so it won't ever complain (and ask us whether or not we actually want to forcefully overwrite conflicts). We could manually diff everything in the workspace to try and determine whether there are conflicts, but that will be a lot slower than just trying `git merge` like we did before (and I'm guessing we don't want to reintroduce using `git merge` either since it comes with its own set of problems w/running DVC in CI).

At this point I don't think it's worth the effort to keep `--no-force`, and would be better to just document how to get back to the user's git state prior to running `exp apply`.

Reply:
Thanks for the explanation. I misread this and thought `--no-force` was still working in this PR. Makes sense.